crates/node/src/command.rs

```diff
@@ -52,69 +52,138 @@ impl SubstrateCli for Cli {
+            cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/testnet-sharingan-raw.json");
+        }
+        (None, Some(Testnet::Sharingan), false, false, _) => {
+            let madara_configs: configs::Configs =
+                serde_json::from_str(&utils::read_file_to_string(madara_path.clone() + "/configs/index.json")?)
+                    .expect("Failed to load madara configs");
+            // If testnet flag is specified, right now it's only Sharingan
+            for chain_spec in madara_configs.chain_specs {
+                configs::fetch_and_validate_file(
+                    madara_configs.remote_base_path.clone(),
+                    chain_spec,
+                    madara_path.clone() + "/configs/chain-specs/",
+                    cli.run.update_configs,
                 )?;
             }
+            cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/testnet-sharingan-raw.json");
         }
+        _ => {}
     }
 
-    // TODO confirm with the CI that we are fetching all and fetch dynamically
-    // Issue #982
-    for file in constants::CAIRO_CONTRACTS_FILES {
-        let src_path = utils::get_project_path();
-        if let Ok(src_path) = src_path {
-            let src_path = src_path + "/configs/cairo-contracts/" + file;
-            utils::copy_from_filesystem(src_path, madara_path.clone() + "/cairo-contracts")?;
-        } else {
-            utils::fetch_from_url(
-                constants::CAIRO_CONTRACTS_URL.to_string() + file,
-                madara_path.clone() + "/cairo-contracts",
+    if cli.run.run_cmd.shared_params.chain.is_some() {
+        cli.run.run_cmd.rpc_external = true;
+        cli.run.run_cmd.rpc_methods = RpcMethods::Unsafe;
+    }
+
+    Ok(())
+}
+
+fn fetch_madara_configs(cli: &mut Cli) -> Result<(), String> {
+    let madara_path = cli.run.madara_path.clone().expect("Failed retrieving madara_path").to_str().unwrap().to_string();
+    let local_path = utils::get_project_path();
+
+    if let Some(configs_url) = cli.run.configs_url.clone() {
+        utils::fetch_from_url(configs_url, madara_path.clone() + "/configs", cli.run.update_configs)?;
+    } else if let Ok(ref src_path) = local_path {
+        let src_path = src_path.clone() + "/configs/index.json";
+        utils::copy_from_filesystem(src_path, madara_path.clone() + "/configs", cli.run.update_configs)?;
+    } else if !cli.run.disable_url_fetch {
+        utils::fetch_from_url(
+            constants::DEFAULT_CONFIGS_URL.to_string(),
+            madara_path.clone() + "/configs",
+            cli.run.update_configs,
+        )?;
+    }
+
+    let madara_configs: configs::Configs =
+        serde_json::from_str(&utils::read_file_to_string(madara_path.clone() + "/configs/index.json")?)
+            .expect("Failed to load madara configs");
+    for asset in madara_configs.genesis_assets {
+        if let Ok(ref src_path) = local_path {
```

Comment:
if we don't plan on handling the error return by `get_project_path`, `local_path` should be an option rather than an Error.

But the true question is : are we absolutely sure we don't want to handle this error?

Reply:
get_project_path will only work in the case we are in a dev environment where we have the cargo project setted. We only look if it's erroring, and if so, we assume that we are not in a dev environment, so we shouldn't try to copy the files

Reply:
We are in the function `fetch_madara_configs`. It's not clear that this function should only be called in a dev env.

I think you should handle the error